ToolBox/GDAXDownloader/GDAXDownloader.cs

```diff
@@ -0,0 +1,114 @@
+        /// <param name="resolution">Only Tick is currently supported</param>
+        /// <param name="startUtc">Start time of the data in UTC</param>
+        /// <param name="endUtc">End time of the data in UTC</param>
+        /// <returns>Enumerable of base data for this symbol</returns>
+        public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            List<BaseData> returnData = new List<BaseData>();
+
+            if (resolution != Resolution.Hour)
+            {
+                throw new ArgumentException("Only hourly data is currently supported.");
+            }
+
+            var counter = startUtc;
+            const string url = "http://api.gdax.com/products/{0}/candles?start={1}&end={2}&granularity={3}";
+
+            while (counter <= endUtc)
+            {
+                System.Threading.Thread.Sleep(1000);
+                Console.WriteLine("Getting " + counter.ToShortDateString() + " data..");
+                DateTime endDate = counter.AddDays(1);
+                int granularity = 3600;
+
+                var requestURL = string.Format(url, symbol.Value, counter.ToShortDateString(), endDate.ToShortDateString(), granularity);
+
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(requestURL);
+                request.UserAgent = ".NET Framework Test Client";
+
+                HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                var encoding = ASCIIEncoding.ASCII;
+
+                string data;
+                using (var reader = new System.IO.StreamReader(response.GetResponseStream(), encoding))
+                {
+                    data = reader.ReadToEnd();
+                }
+
+                data = data.Remove(0, 2);
+                data = data.Substring(0,data.Length - 2);
+
+                string[] bars = data.Split(new string[] { "],[" },StringSplitOptions.RemoveEmptyEntries);
+
+                foreach(var bar in bars.Distinct())
+                {
+                    var items = bar.Split(new char[] { ',' });
+                    TradeBar tradeBar = new TradeBar()
+                    {
+                        Time =  new DateTime(1970, 1, 1, 0, 0, 0, 0).AddSeconds(long.Parse(items[0])).ToLocalTime(),
+                        Symbol = symbol,
+                        Low = decimal.Parse(items[1]),
+                        High = decimal.Parse(items[2]),
+                        Open = decimal.Parse(items[3]),
+                        Close = decimal.Parse(items[4]),
+                        Volume = decimal.Parse(items[5]),
+                        Value = decimal.Parse(items[4]),
+                        DataType = MarketDataType.TradeBar,
+                        Period = Time.OneHour,
+                        EndTime = new DateTime(1970, 1, 1, 0, 0, 0, 0).AddSeconds(long.Parse(items[0])).ToLocalTime().AddHours(1)
```

Comment:
I think there's helpers in Time class for creating a DateTime from Unix

Reply:
I only find such helper in .Net 4.6 
http://gigi.nullneuron.net/gigilabs/converting-tofrom-unix-timestamp-in-c/

Reply:
@espirulina I think Jared meant `QuantConnect.Time` static class:
https://github.com/QuantConnect/Lean/blob/master/Common/Time.cs#L128